### PR TITLE
OCM-7309 | feat: update kubeletconfig resource to support HCP

### PR DESCRIPTION
**Pull request by marcolan018**
Signed-off-by: marcolan018 <llan@redhat.com>

**What this PR does / why we need it**:
this pr updates kubeletconfig resource to support HCP.  Some points for this change:

- the single resource to handle both classic&hcp clusters
- updates the existing resource to use the HCP endpoint

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story (OCM-xxxx)*:
Fixes #[OCM-7309](https://issues.redhat.com//browse/OCM-7309),[OCM-7310](https://issues.redhat.com/browse/OCM-7310)

**Change type**
- [x] New feature
- [ ] Bug fix
- [ ] Build
- [ ] CI
- [ ] Documentation
- [ ] Performance
- [ ] Refactor
- [ ] Style
- [ ] Unit tests
- [ ] Subsystem tests

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.

### CODE REVIEW

File: provider/kubeletconfig/kubeletconfig_resource.go [hunk 1]
```diff
@@ -163,10 +199,25 @@ func (k *KubeletConfigResource) Read(ctx context.Context, req resource.ReadReque
 	}
 
 	clusterId := state.Cluster.ValueString()
-	exists, kubeletConfig, err := k.configClient.Exists(ctx, clusterId)
+
+	var kubeletConfigId string
+	var err error
+	if common.IsStringAttributeUnknownOrEmpty(state.ID) {
```
Comment (den-rgb) [inline provider/kubeletconfig/kubeletconfig_resource.go:205]: Could we move this check into a common function since its being used in 3 different instances?

File: provider/kubeletconfig/kubeletconfig_resource.go [hunk 2]
```diff
@@ -264,7 +350,44 @@ func (k *KubeletConfigResource) Configure(_ context.Context, req resource.Config
...
+	}
+	// for classic cluster, there is maxium one kubeletconfig
+	if name == "" {
+		if len(kubeletConfigs) > 0 {
+			return kubeletConfigs[0].ID(), nil
+		} else {
+			return id, fmt.Errorf("Cannot find KubeletConfig for cluster '%s'", clusterId)
+		}
+	}
+	for _, config := range kubeletConfigs {
```
Comment (den-rgb) [inline provider/kubeletconfig/kubeletconfig_resource.go:383]: Is this loop necessary since there'll only ever be one kubeletConfig anyway?

File: provider/machinepool/hcp/machine_pool_resource.go [hunk 3]
```diff
@@ -180,6 +180,10 @@ func (r *HcpMachinePoolResource) Schema(ctx context.Context, req resource.Schema
 				ElementType: types.StringType,
 				Optional:    true,
 			},
+			"kubelet_configs": schema.StringAttribute{
+				Description: "Name of the kubelet config applied to the machine pool. A single kubelet config is allowed. Kubelet config must already exist.",
```
Comment (den-rgb) [inline provider/machinepool/hcp/machine_pool_resource.go:184]: Maybe reword it to " A single kubelet config can be applied at a time" instead of "allowed" since HCP clusters can hold more than one.

File: provider/kubeletconfig/kubeletconfig_resource.go (hunk 2 above)
Comment (marcolan018) [reply provider/kubeletconfig/kubeletconfig_resource.go:383]: for one HCP cluster, there maybe more than one kubeletConfig. one machinepool can only have single one.

File: provider/kubeletconfig/kubeletconfig_resource.go (hunk 1 above)
Comment (marcolan018) [reply provider/kubeletconfig/kubeletconfig_resource.go]: made the refactoring, please re-review again, thanks

File: subsystem/kubeletconfig_resource_test.go [hunk 4]
```diff
@@ -82,6 +82,20 @@ var _ = Describe("Cluster KubeletConfig", func() {
 
 		It("fails if OCM backend fails to create the object", func() {
```
Comment (robpblake) [inline subsystem/kubeletconfig_resource_test.go:83]: @marcolan018 We should extend this test to cover HCP clusters as well, particularly ensuring we allow the user to create multiple kubeletconfigs.

**Comment by robpblake** [comment]
@marcolan018 Can you also please update https://github.com/terraform-redhat/terraform-provider-rhcs/blob/main/subsystem/hcp/machine_pool_resource_test.go to ensure we can create/edit the `kubelet_configs` field on a HCP Machine Pool.

File: subsystem/kubeletconfig_resource_test.go (hunk 4 above)
Comment (marcolan018) [reply subsystem/kubeletconfig_resource_test.go:293]: new subsystem tests added for kubelet configs

**Review by robpblake — APPROVED** [review]
/lgtm
